libcorsaro/libcorsaro_filtering.c

```diff
@@ -368,6 +368,18 @@ static inline int _apply_port_tcp5000_filter(corsaro_logger_t *logger,
     return 0;
 }
 
+static inline int _apply_tcp_port_1433_scan_filter(corsaro_logger_t *logger,
+        filter_params_t *fparams) {
+    if (!fparams->ip)
+        return -1;
+
+    if (fparams->tcp != NULL && fparams->tcp->syn && fparams->dest_port == 1433 && fparams->ip->ip_ttl < 64 && fparams->tcp->window == 32 && fparams->ip->ip_len == 11264) {
```

Comment:
1. Rather than having 1 big logical `if` condition, it would be nicer to split this into a bunch of smaller `if` checks, e.g.
```
if (fparams->tcp == NULL) {
    return 0;
}
if (fparams->ip->ip_ttl >= 64) {
    return 0;
}
... remaining conditions here ...

return 1;
```

2. Use `ntohs()` to convert 16-bit values that are being read directly from a TCP or IP header into the native byte order for the CPU. Then your comparisons can be done against the "real" value intended by the packet sender, which also means your code is a bit more self-documenting. E.g.:
```
if (ntohs(fparams->ip->ip_len) != 44) {   // because the IP length is actually 44 bytes, not 11264 bytes
    return 0;
}
```

3. Whenever you want to match on only SYN packets, make sure you also check for the absence of the ACK flag -- otherwise your rule will also match SYN-ACK packets:
```
if ((!fparams->tcp->syn) || fparams->tcp->ack) {
    return 0;
}
```